src/test/java/de/qabel/desktop/AsyncUtils.java

```diff
@@ -32,4 +34,36 @@ public static void waitUntil(Callable<Boolean> evaluate, long timeout, Callable<
             fail(e.getMessage());
         }
     }
+
+    public static void runLaterAndWait(Runnable runnable) {
+        boolean[] hasRun = new boolean[]{false};
+        Platform.runLater(() -> {
+            runnable.run();
+            hasRun[0] = true;
+        });
+        waitUntil(() -> hasRun[0], 5000L);
+    }
+
+    public static void assertAsync(AbstractController.CheckedRunnable assertion) {
+        assertAsync(assertion, 2000L);
+    }
+
+    public static void assertAsync(AbstractController.CheckedRunnable assertion, long timeout) {
+        long startTime = System.currentTimeMillis();
+        while (true) {
+            try {
+                Thread.yield();
+                Thread.sleep(10);
+                assertion.run();
+                return;
+            } catch(AssertionError e){
+                if (System.currentTimeMillis() - timeout > startTime) {
+                    throw e;
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
The fail only preserves the message. So we need a Stacktrace from anywhere. And in test code I see no point in roundtripping through a logger, do you?